code/game/objects/structures/window_spawner.dm

```diff
@@ -35,64 +35,46 @@
 		activate()
 		return INITIALIZE_HINT_QDEL
 
-/obj/effect/wingrille_spawn/proc/activate()
-	if(activated) return
-	if (!locate(/obj/structure/grille) in get_turf(src))
-		var/obj/structure/grille/G = PoolOrNew(/obj/structure/grille, src.loc)
-		handle_grille_spawn(G)
-	var/list/neighbours = list()
-	for (var/dir in cardinal)
-		var/turf/T = get_step(src, dir)
-		var/obj/effect/wingrille_spawn/other = locate(/obj/effect/wingrille_spawn) in T
-		if(!other)
-			var/found_connection
-			if(locate(/obj/structure/grille) in T)
-				for(var/obj/structure/window/W in T)
-					if(W.type == win_path && W.dir == get_dir(T,src))
-						found_connection = 1
-						qdel(W)
-			if(!found_connection)
-				var/obj/structure/window/new_win = PoolOrNew(win_path, src.loc)
-				new_win.set_dir(dir)
-				handle_window_spawn(new_win)
-		else
-			neighbours |= other
-	activated = 1
-	for(var/obj/effect/wingrille_spawn/other in neighbours)
-		if(!other.activated) other.activate()
-
-/obj/effect/wingrille_spawn/proc/handle_window_spawn(var/obj/structure/window/W)
+/obj/effect/window_lwall_spawn/proc/handle_window_spawn(var/obj/structure/window/W)
+	PoolOrNew(win_path, src.loc)
 	return
 
-// Currently unused, could be useful for pre-wired electrified windows.
-/obj/effect/wingrille_spawn/proc/handle_grille_spawn(var/obj/structure/grille/G)
+/obj/effect/window_lwall_spawn/proc/activate()
+	PoolOrNew(/obj/structure/grille, src.loc)
+	handle_window_spawn(src)
+	activated = TRUE
 	return
 
-/obj/effect/wingrille_spawn/reinforced
+/obj/effect/window_lwall_spawn/reinforced
 	name = "reinforced window grille spawner"
-	icon_state = "r-wingrille"
-	win_path = /obj/structure/window/reinforced
+	icon_state = "sp_full-window"
+	win_path = /obj/structure/window/reinforced/full
 
-/obj/effect/wingrille_spawn/reinforced/crescent
-	name = "Crescent window grille spawner"
-	win_path = /obj/structure/window/reinforced/crescent
+/obj/effect/window_lwall_spawn/smartspawn
+	name = "reinforced window smart spawner"
+	icon_state = "sp-smart_full-window"
 
-/obj/effect/wingrille_spawn/plasma
-	name = "plasma window grille spawner"
-	icon_state = "p-wingrille"
-	win_path = /obj/structure/window/plasmabasic
+/obj/effect/window_lwall_spawn/smartspawn/handle_window_spawn(var/obj/structure/window/W)
+	if ((locate(/turf/space) in range(1, src)) || (locate(/turf/simulated/floor/hull) in range(1, src)))
+		PoolOrNew(/obj/structure/window/reinforced/full, src.loc)
+	else
+		PoolOrNew(/obj/structure/window/basic/full, src.loc)
+	return
 
-/obj/effect/wingrille_spawn/reinforced_plasma
-	name = "reinforced plasma window grille spawner"
-	icon_state = "pr-wingrille"
-	win_path = /obj/structure/window/reinforced/plasma
+/obj/effect/window_lwall_spawn/plasma
+	name = "plasma window grille spawner"
+	icon_state = "sp_full-window-plasma"
+	win_path = /obj/structure/window/plasmabasic/full
 
-/obj/effect/wingrille_spawn/reinforced/polarized
+/obj/effect/window_lwall_spawn/reinforced/polarized
```

Comment:
are tehse properly placed in the map so that places like the brig will always have reinforced windows?

Reply:
Yeah, brig got those spawners.